plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetsPage/DatasetTree/index.tsx

```diff
@@ -43,24 +44,12 @@ function typeFilter(catalog: CatalogRawItem[]): DataCatalog {
   return filteredCatalog;
 }
 
-function tagFilter(catalog: CatalogRawItem[]): DataCatalog {
-  const filteredCatalog: CatalogItem[] = prefectures.map(p => {
-    const items: CatalogItem[] = catalog.filter(i => {
-      if (i.prefecture === p) {
-        return {
-          type: "item",
-          ...i,
-        };
-      }
-    }) as CatalogItem[];
-
-    return {
-      type: "group",
-      name: p,
-      children: items,
-    };
-  });
-  return filteredCatalog;
+function tagFilter(catalog: CatalogRawItem[], tags?: Tag[]): DataCatalog {
+  return catalog
+    .filter(item =>
+      item.tags?.some(tag => tags?.some(selectedTag => tag.name === selectedTag.name)),
```

Comment:
I could be wrong, but can you simplify this by doing:
```
return catalog
    .filter(item =>
      item.tags?.some(tag => tags?.includes(tag)),
```

Reply:
This might work if `tag` was of type string (strings are passed by value in javascript). 
However, since `tag` is an object (objects are passed by reference in javascript) the `.includes()` method would not work.